docs/source/dev/py_postprocessing.rst

```diff
@@ -0,0 +1,178 @@
+.. _development-pytools:
+
+Python postprocessing tool structure
+====================================
+
+For each plugin there are (or have to be implemented) three different classes.
+
+1. A data reader class responsible for loading the data from the simulation directory
+2. A visualizer class that outputs a matplotlib plot
+3. A jupyter-widget class that exposes the parameters of the matplotlib visualizer to the user
+   via other widgets
+
+Data reader
+~~~~~~~~~~~
+The data readers should reside in the ``lib/python/picongpu/plugins/`` directory.
+
+Each reader class needs to implement the following interface functions:
+
+.. code:: python
+
+   def __init__(self, run_directory)
+
+   def get_data_path(self, your_specific_arguments)
+       # has to return the path to the underlying data file
+       # specified by some specific arguments (depending on the plugin output)
+
+   def get_iterations(self, your_specific_arguments)
+       # return a numpy array with iteration numbers for which data is available
+
+   def get(self, your_specific_arguments)
+       # main function of the reader to actually get the data
+
+Visualizer
+~~~~~~~~~~
+The visualizers should reside in the ``lib/python/picongpu/plugins/plot_mpl/`` directory.
+The module names should end on ``_visualizer.py`` and the class name should only be ``Visualizer``.
+
+There is a base class for visualization found in ``base_visualizer.py`` which already handles the plotting logic. It uses the
+data reader classes for accessing the data.
+After getting the data, it ensures that (for performance reasons) a matplotlib artist is created only for the first plot and later only gets updated with fresh data.
+
```

Comment:
I think we want to include the base_visualizer docstrings here via
```rst
.. autoclass:: picongpu.plugins.plot_mpl.base_visualizer.Visualizer
```

Reply:
This had no effect when I built the doc.

Reply:
Yes, I guess we need to set the PYTHONPATH first... will check

Reply:
added now with last commit to avoid updating the interface in the manual *and* the source (now source only :) )